### PR TITLE
remove references to v1beta1/2 in listwatch_test.go

### PR DESCRIPTION
**Pull request by caesarxuchao**
#8087

@nikhiljindal @bgrant0607 @krousey

### CODE REVIEW

File: pkg/client/cache/listwatch_test.go [hunk 1]
```diff
@@ -87,7 +63,7 @@ func TestListWatchesCanList(t *testing.T) {
 	}{
 		// Minion
 		{
-			location:      buildLocation(buildResourcePath("", api.NamespaceAll, "minions"), buildQueryValues(api.NamespaceAll, nil)),
+			location:      testapi.ResourcePathWithPrefix("", "minions", api.NamespaceAll, ""),
```
Comment (nikhiljindal) [inline pkg/client/cache/listwatch_test.go]: You can use ResourcePath since the prefix is empty.

File: pkg/client/cache/listwatch_test.go [hunk 2]
```diff
@@ -95,20 +71,20 @@ func TestListWatchesCanList(t *testing.T) {
 		// pod with "assigned" field selector.
 		{
 			location: buildLocation(
-				buildResourcePath("", api.NamespaceAll, "pods"),
-				buildQueryValues(api.NamespaceAll, url.Values{fieldSelectorQueryParamName: []string{getHostFieldLabel() + "="}})),
+				testapi.ResourcePathWithPrefix("", "pods", api.NamespaceAll, ""),
```
Comment (nikhiljindal) [inline pkg/client/cache/listwatch_test.go]: Use ResourcePath

File: pkg/client/cache/listwatch_test.go [hunk 3]
```diff
@@ -95,20 +71,20 @@ func TestListWatchesCanList(t *testing.T) {
...
 			namespace:     api.NamespaceAll,
-			fieldSelector: fields.Set{getHostFieldLabel(): ""}.AsSelector(),
+			fieldSelector: fields.Set{"spec.host": ""}.AsSelector(),
 		},
 		// pod in namespace "foo"
 		{
 			location: buildLocation(
-				buildResourcePath("", "foo", "pods"),
-				buildQueryValues("foo", url.Values{fieldSelectorQueryParamName: []string{getHostFieldLabel() + "="}})),
+				testapi.ResourcePathWithPrefix("", "pods", "foo", ""),
```
Comment (nikhiljindal) [inline pkg/client/cache/listwatch_test.go]: Use ResourcePath

**Comment by nikhiljindal** [comment]
LGTM, modulo the comments.

**Comment by caesarxuchao** [comment]
Updated. PTAL. Thank you!

**Comment by nikhiljindal** [comment]
LGTM, thanks!

**Pull request merged**